src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOfTests.cs

```diff
@@ -0,0 +1,810 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. 
+
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.PreferStringContainsOverIndexOfAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.PreferStringContainsOverIndexOfFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.PreferStringContainsOverIndexOfAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.PreferStringContainsOverIndexOfFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class PreferStringContainsOverIndexOfTests
+    {
+        [Theory]
+        [InlineData("This", "This", false, " == ", " -1", "!")]
+        [InlineData("a", "a", true, " == ", " -1", "!")]
+        [InlineData("This", "This", false, " >= ", " 0", "")]
+        [InlineData("a", "a", true, " >= ", " 0", "")]
+        public async Task TestStringAndChar(string input, string fix, bool isCharTest, string operatorKind, string value, string notString)
+        {
+            string startQuote = isCharTest ? "'" : "\"";
+            string endQuote = isCharTest ? "'" : "\", System.StringComparison.Ordinal";
+            string csInput = @"
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            const string str = ""This is a string"";
+            int index = str.IndexOf(" + startQuote + input + endQuote + @");
+            if ([|index" + operatorKind + value + @"|])
```

Comment:
What if I had intermediate code that re-assigned either `str` or the argument passed to `IndexOf` call? As in,

```csharp
string str = "First string";
int index = str.IndexOf("First");
str = "Second string";
if (index != -1)
{
   // This code should execute
}
```

OR

```csharp
string str = "First string";
string subStringToFind = "First";
int index = str.IndexOf(subStringToFind);
subStringToFind = "Second";
if (index != -1)
{
   // This code should execute
}
```

I think your diagnostic/code fix will cause the user's code to break at runtime.

Reply:
Good catch. I'll fix these. I'm also addressing cases where `str` is a class field or a string literal.